docs/user-guide/data_types.rst

```diff
@@ -1,33 +1,54 @@
 
-.. code-block:: python
+Zarr arrays operate under an essential design constraint: Unlike NumPy arrays, Zarr arrays
+are designed to stored and accessed by other Zarr implementations. This means that, among other things,
+Zarr data types must be serializable to metadata documents in accordance with the Zarr specifications,
+and this adds some unique texture to the Zarr data type model.
 
-  >>> import zarr
-  >>> import numpy as np
-  >>> z = zarr.create_array(store={}, shape=(10,), dtype=np.dtype('uint8'))
-  >>> z
-  <Array memory:... shape=(10,) dtype=uint8>
+The following sections will explain Zarr's data type model in greater detail, and demonstrate the
+Zarr Python APIs for working with Zarr data types.
+
+Array data types
+^^^^^^^^^^^^^^^^
+
+Every Zarr array has a data type, which defines the meaning of the array's elements. An array's data
+type is encoded in the metadata for the array, which means that a declaration of the
+data type of an array must be `JSON`-serializable.
+
+In Zarr V2, the data type of an array is stored in the ``dtype`` field in array metadata.
+Zarr V3 changed the name of this field to ``data_type``, and also defined new rules for the values
+assignable to the ``data_type`` field.
+
+For example, in Zarr V2, the boolean array data type was represented in array metadata as the
+string ``"|b1"``. In Zarr V3, the same type is represented as the string ``"bool"``.
+
+Scalars
+^^^^^^^
+
+Zarr also defines how array elements, i.e. scalars, are encoded in array metadata. This is necessary
+because Zarr uses a field in array metadata to define a default value for chunks that are not stored.
+This field, called ``fill_value`` in both Zarr V2 and Zarr V3 metadata documents, contains a
+``JSON`` value that can be decoded to a scalar value compatible with the array's data type.
+
+For the boolean data type, the scalar encoding is simple -- booleans are natively supported by
+``JSON``, so zarr saves booleans as ``JSON`` booleans. Other scalars, like floats or raw bytes, have
+more elaborate encoding schemes, and in some cases this scheme depends on the Zarr format version.
 
-Unlike NumPy arrays, Zarr arrays are designed to accessed by Zarr
-implementations in different programming languages. This means Zarr data types must be interpreted
-correctly when clients read an array. Each Zarr data type defines procedures for
-encoding and decoding both the data type itself, and scalars from that data type to and from Zarr array metadata. And these serialization procedures
-depend on the Zarr format.
 
 Data types in Zarr version 2
 -----------------------------
 
 Version 2 of the Zarr format defined its data types relative to
 `NumPy's data types <https://numpy.org/doc/2.1/reference/arrays.dtypes.html#data-type-objects-dtype>`_,
-and added a few non-NumPy data types as well. Thus the JSON identifier for a NumPy-compatible data
+and added a few non-NumPy data types as well. With one exception, the Zarr V2 JSON identifier for a data
```

Comment:
```suggestion
and added a few non-NumPy data types as well. With one exception (`<the exception>`), the Zarr V2 JSON identifier for a data
```

otherwise confusing with the mention of two special cases below

Reply:
done